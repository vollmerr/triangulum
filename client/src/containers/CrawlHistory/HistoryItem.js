import React from 'react';
import { List } from 'semantic-ui-react';

const HistoryItem = item => (
  <List.Item>
    <List.Content floated='right'>
      <List.Description>{new Date(item.updated).toUTCString()}</List.Description>
    </List.Content>
    <List.Content>
      <List.Header>{item.request.url}</List.Header>
      <List.Description>{`Type: ${item.request.type}`}</List.Description>
      <List.Description>{`Limit: ${item.request.limit}`}</List.Description>
      {
        item.request.keyword
        && <List.Description>{`Keyword: ${item.request.keyword}`}</List.Description>
      }
    </List.Content>
  </List.Item>
);

export default HistoryItem;
