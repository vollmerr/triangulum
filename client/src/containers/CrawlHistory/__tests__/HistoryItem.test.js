import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'semantic-ui-react';

import HistoryItem from '../HistoryItem';

const props = {
  updated: new Date('2018-01-02'),
  request: {
    url: 'http://test.com',
    type: 'bfs',
    limit: 100,
    keyword: 'test keyword',
  },
};

let wrapper;
describe('HistoryItem', () => {
  beforeEach(() => {
    wrapper = shallow(<HistoryItem {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render not a keyword if none provided', () => {
    wrapper.setProps({ request: { ...props.request, keyword: null } });
    const keywordNode = wrapper
      .find(List.Description)
      .findWhere(x => x.text().match(/Keyword/));
    expect(keywordNode.length).toEqual(0);
  });
});
