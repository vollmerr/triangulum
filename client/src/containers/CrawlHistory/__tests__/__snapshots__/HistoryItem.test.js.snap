// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem should render correctly 1`] = `
<ListItem>
  <ListContent
    floated="right"
  >
    <ListDescription>
      Tue, 02 Jan 2018 00:00:00 GMT
    </ListDescription>
  </ListContent>
  <ListContent>
    <ListHeader>
      http://test.com
    </ListHeader>
    <ListDescription>
      Type: bfs
    </ListDescription>
    <ListDescription>
      Limit: 100
    </ListDescription>
    <ListDescription>
      Keyword: test keyword
    </ListDescription>
  </ListContent>
</ListItem>
`;
